### PR TITLE
Migrate localized-multiline-text-input to typescript

### PR DESCRIPTION
**Pull request by ddouglasz**
Migrate localized-multiline-text-input to typescript

### CODE REVIEW

File: packages/components/inputs/localized-multiline-text-input/src/translation-input.styles.ts [hunk 1]
```diff
@@ -9,7 +9,12 @@ import {
 // * a disabled-field currently does not display warning/error-states so it takes precedence
 // * a readonly-field cannot be changed, but it might be relevant for validation, so error and warning are checked first
 // how you can interact with the field is controlled separately by the props, this only influences visuals
-const getTextareaStyles = (props) => {
+
+type TTranslateInputStylesProps = {
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/localized-multiline-text-input/src/translation-input.styles.ts:13]: Could we use the name based on this filename? 
I was expecting it to be `TTranslationInputStylesProps`

File: packages/components/inputs/localized-multiline-text-input/src/localized-multiline-text-input.tsx [hunk 2]
```diff
@@ -125,11 +255,15 @@ const LocalizedMultilineTextInput = (props) => {
               <TranslationInput
                 key={language}
                 autoComplete={props.autoComplete}
-                id={LocalizedMultilineTextInput.getId(props.id, language)}
+                id={LocalizedMultilineTextInput.getId(
+                  props.id,
+                  language as string
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/localized-multiline-text-input/src/localized-multiline-text-input.tsx:260]: I'm wondering why we need to explicitly say `language` is a string when here we're iterating over the `languages` constants which has been actually defined as an array of strings.

File: packages/components/inputs/localized-multiline-text-input/docs/usage-example.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    value={{ en: 'House\nFoo', de: 'House' }}
+    onChange={
+      (/** event */) => {
+        // alert(event.target.name, event.target.value)
+      }
+    }
+  />
+);
+
+export default Example;
```
Comment (emmenko) [inline packages/components/inputs/localized-multiline-text-input/docs/usage-example.tsx:15]: This file is expected to be a `.js` file

File: packages/components/inputs/localized-multiline-text-input/src/translation-input.tsx [hunk 4]
```diff
@@ -64,15 +99,15 @@ const TranslationInput = (props) => {
...
+      onChange && onChange(event);
     },
     [onChange, props.language]
   );
 
   const { onFocus, onToggle } = props;
   const handleFocus = useCallback(() => {
     // Expand the input on focus
-    if (props.isCollapsed) onToggle();
+    if (props.isCollapsed) onToggle && onToggle();
```
Comment (emmenko) [inline packages/components/inputs/localized-multiline-text-input/src/translation-input.tsx:110]:
```suggestion
    if (props.isCollapsed) onToggle?.();
```

**Review by emmenko — COMMENTED** [review]
Looks good overall, just some minor improvements.

File: packages/components/inputs/localized-multiline-text-input/src/translation-input.tsx [hunk 5]
```diff
@@ -1,12 +1,15 @@
-import { useState, useCallback } from 'react';
-import PropTypes from 'prop-types';
-import requiredIf from 'react-required-if';
+import {
+  useState,
+  useCallback,
+  ChangeEventHandler,
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/localized-multiline-text-input/src/translation-input.tsx:4]:
```suggestion
  type ChangeEventHandler,
```

**Review by CarlosCortizasCT — CHANGES_REQUESTED** [review]

**Review by CarlosCortizasCT — APPROVED** [review]
LGTM

**Review by emmenko — APPROVED** [review]

**Pull request merged**